tap_dynamodb/sync_strategies/log_based.py

```diff
@@ -115,6 +115,27 @@ def sync_shard(shard, seq_number_bookmarks, streams_client, stream_arn, projecti
     singer.write_state(state)
     return rows_synced
 
+def prepare_projection(projection, expression, exp_key_traverse):
+    '''
+    Prepare the projection based on the expression attributes
+    For example:
+    projections = "#c", expressions = {"#c": "Comment"}
+    return = ["Comment"]
+    Example of nested expression :
+    In Catalog - projections = "#n[0].#a", expressions = {"#n": "Name", "#a": "Age"}
+    projection passed in the function = ['#n[0]', '#a']
+    projection returned from the function = ['Name[0]', 'Age']
+    '''
+    for i, part in enumerate(projection):
+        if '#' in part:
+            replaceable_part = part.split('[', 1)[0] if '[' in part else part
+            # check if the projection placeholder is defined in the expression else raise an exception
+            if replaceable_part in expression:
+                exp_key_traverse.discard(replaceable_part)
+                # replace the value given in the expression with the key in the projection
+                projection[i] = part.replace(replaceable_part, expression[replaceable_part])
+            else:
+                raise Exception("No expression is available or the given projection: {}.".format(replaceable_part))
```

Comment:
```suggestion
                raise Exception("No expression is available for the given projection: {}.".format(replaceable_part))
```
Also, update the same message in a unit test.

Reply:
Changed the error message, also updated the unittest